src/checkbox.test.mouse.ts

```diff
@@ -0,0 +1,234 @@
+
+  await expect(host).toDispatchEvents(
+    () => page.getByRole('checkbox').click(),
+    [
+      {
+        bubbles: true,
+        cancelable: false,
+        composed: true,
+        defaultPrevented: false,
+        type: 'input',
+      },
+      {
+        bubbles: true,
+        cancelable: false,
+        composed: true,
+        defaultPrevented: false,
+        type: 'change',
+      },
+    ],
+  );
+
+  await expect(host).toHaveJSProperty('checked', true);
+});
+
+test(
+  'can be checked via `click()`',
+  { tag: '@mouse' },
+  async ({ callMethod, mount, page }) => {
+    await mount(
+      html`<glide-core-checkbox label="Label"></glide-core-checkbox>`,
+    );
+
+    const host = page.locator('glide-core-checkbox');
+
+    await expect(host).toDispatchEvents(
+      () => callMethod(host, 'click'),
+      [
+        {
+          bubbles: true,
+          cancelable: false,
+          composed: true,
+          defaultPrevented: false,
+          type: 'input',
+        },
+        {
+          bubbles: true,
+          cancelable: false,
+          composed: true,
+          defaultPrevented: false,
+          type: 'change',
+        },
+      ],
+    );
+
+    await expect(host).toHaveJSProperty('checked', true);
+  },
+);
+
+test(
+  'can be unchecked via mouse',
+  { tag: '@mouse' },
+  async ({ mount, page }) => {
+    await mount(
+      html`<glide-core-checkbox label="Label" checked></glide-core-checkbox>`,
+    );
+
+    const host = page.locator('glide-core-checkbox');
+
+    await expect(host).toDispatchEvents(
+      () => page.getByRole('checkbox').click(),
+      [
+        {
+          bubbles: true,
+          cancelable: false,
+          composed: true,
+          defaultPrevented: false,
+          type: 'input',
+        },
+        {
+          bubbles: true,
+          cancelable: false,
+          composed: true,
+          defaultPrevented: false,
+          type: 'change',
+        },
+      ],
+    );
+
+    await expect(host).toHaveJSProperty('checked', false);
+  },
+);
+
+test(
+  'can be unchecked via `click()`',
+  { tag: '@mouse' },
+  async ({ callMethod, mount, page }) => {
+    await mount(
+      html`<glide-core-checkbox label="Label" checked></glide-core-checkbox>`,
+    );
+
+    const host = page.locator('glide-core-checkbox');
+
+    await expect(host).toDispatchEvents(
+      () => callMethod(host, 'click'),
+      [
+        {
+          bubbles: true,
+          cancelable: false,
+          composed: true,
+          defaultPrevented: false,
+          type: 'input',
+        },
+        {
+          bubbles: true,
+          cancelable: false,
+          composed: true,
+          defaultPrevented: false,
+          type: 'change',
+        },
+      ],
+    );
+
+    await expect(host).toHaveJSProperty('checked', false);
+  },
+);
+
+test(
+  'checked and not indeterminate when unchecked and indeterminate and clicked via mouse',
```

Comment:
Open to suggestions, of course. I'm still working through how to name tests without our beloved `it()`.